### PR TITLE
Add config param to enable/disable wallet signature

### LINKED ISSUE #468: Add config param to enable/disable wallet signature
Depends on https://github.com/ubiquity/ubiquibot/issues/410

We have a [feature](https://github.com/ubiquity/ubiquibot/issues/410) of registering a wallet with a provided signature to ensure that bounty hunter has a private key (i.e. wallet address is not a CEX address) in order to claim a reward.

We should have the ability to enable/disable the "register wallet with signature" feature.

What should be done:
- Add a new bot's config property for enabling/disabling the "register wallet with verified signature" feature. The property should be `false` by default
- If the `register-wallet-with-verification` property is set to `true` then bounty hunters must be forced to register their wallets only with the provided signature ([example](https://github.com/ubiquity/ubiquibot/issues/410#issuecomment-1614202635))
- If the `register-wallet-with-verification` property is set to `false` then bounty hunters can register without a signature ([example](https://github.com/ubiquity/ubiquibot/issues/315#issuecomment-1577038350))

_Originally posted by @pavlovcik in https://github.com/ubiquity/ubiquibot/issues/410#issuecomment-1621438825_

### PR DESCRIPTION
**Pull request by web4er**
Resolves #468

### CODE REVIEW

**Comment by 0xcodercrane** [comment]
You need to post a QA link @web4er

**Comment by web4er** [comment]
> You need to post a QA link @web4er

It is mentioned above. Here it is again. https://github.com/web4er/ubiquibot/issues/21

File: src/bindings/config.ts [hunk 1]
```diff
@@ -68,6 +74,9 @@ export const loadConfig = async (context: Context): Promise<BotConfig> => {
       privateKey: process.env.X25519_PRIVATE_KEY ?? "",
       publicKey: publicKey ?? "",
     },
+    wallet: {
+      registerWalletWithVerification: process.env.REGISTER_WALLET_WITH_VERIFICATION?.toLowerCase() === "true" ?? DEFAULT_REGISTER_WALLET_WITH_VERIFICATION,
```
Comment (0xcodercrane) [inline src/bindings/config.ts:78]: We should make the `registerWalletWithVerification` as an org/repo-specific variable, not the bot-specific variable.

What needs to be updated for that?
1/ Read the value from `getWideConfig` and set the value like `baseMultiplier`/`issueCreatorMultiplier`.
2/ Remove `DEFAULT_REGISTER_WALLET_WITH_VERIFICATION`

Comment (web4er) [reply src/bindings/config.ts]: This is done and here is a new QA after making this fix https://github.com/web4er/ubiquibot/issues/25

**Comment by 0xcodercrane** [comment]
@web4er would you resolve the conflicts?

**Comment by web4er** [comment]
> @web4er would you resolve the conflicts?

I have handled the conflicts and you should be able to merge now. 

I had to merge twice, because first I merged and re-tested it becasue conflicts were a bit extensive. In this duration more changes got pushed to 'development' branch lol. So had to merge and test again. 

Here is an updated QA https://github.com/web4er/ubiquibot/issues/44

@0xcodercrane

**Comment by rndquu** [comment]
> > @web4er would you resolve the conflicts?
> 
> I have handled the conflicts and you should be able to merge now.
> 
> I had to merge twice, because first I merged and re-tested it becasue conflicts were a bit extensive. In this duration more changes got pushed to 'development' branch lol. So had to merge and test again.
> 
> Here is a new QA [web4er#44](https://github.com/web4er/ubiquibot/issues/44)
> 
> @0xcodercrane

QA is structured perfectly

**Review by rndquu — APPROVED** [review]

**Comment by rndquu** [comment]
@web4er pls fix the conflicts again, the PR should be merged automatically when conflicts are resolved

**Comment by pavlovcik** [comment]
I think we should consider some type of incentive structure for bounty hunters to keep fixing merge conflicts.

**Comment by web4er** [comment]
I have handled the conflicts and you should be able to merge now.

> @web4er pls fix the conflicts again, the PR should be merged automatically when conflicts are resolved

It appears the auto-merge is blocked because I resolve merged conflicts locally, and not on github. @rndquu 


> I think we should consider some type of incentive structure for bounty hunters to keep fixing merge conflicts.

This PR changes file `src/handlers/comment/handlers/index.ts` which is contantly being changed. That is why almost every other merge is creating a conflict. Solution here is to merge soon.

**Review by rndquu — APPROVED** [review]

**Review by 0xcodercrane — APPROVED** [review]

**Comment by 0xcodercrane** [comment]
It doesn't require any configuration changes in staging/production bot? @rndquu

**Comment by rndquu** [comment]
> It doesn't require any configuration changes in staging/production bot? @rndquu

No, the default `false` value will be used for the `register-wallet-with-verification` param unless it is explicitly enabled in one of the configs

**Pull request merged**